dask/dataframe/core.py

```diff
@@ -5054,8 +5054,17 @@ def eval(self, expr, inplace=None, **kwargs):
         meta = self._meta.eval(expr, inplace=inplace, **kwargs)
         return self.map_partitions(M.eval, expr, meta=meta, inplace=inplace, **kwargs)
 
+    if PANDAS_GT_150:
+        from pandas._libs.lib import no_default
+
+        how_default = no_default
+        thresh_default = no_default
+    else:
+        how_default = "any"
+        thresh_default = None
+
     @derived_from(pd.DataFrame)
-    def dropna(self, how="any", subset=None, thresh=None):
+    def dropna(self, how=how_default, subset=None, thresh=thresh_default):
```

Comment:
Currently we're trying to match `pandas` behavior here exactly, which is normally a guiding philosophy. However, in this particular case, since the change in `pandas` was to fix a bug where incompatible keywords could be specified and unexpected results returned https://github.com/pandas-dev/pandas/issues/46575, I think it would make sense for us to always raise an error when `how` and `thresh` are both specified (i.e. always match the new `pandas` behavior). While this does lead to slightly different behavior for older versions of  `pandas`, Dask users would still benefit from not having an unexpected result returned

Reply:
@jrbourbeau Should we then set both to `no_default` and handle manually the case pandas < 1.5. following what pandas does, something like

```python
    @derived_from(pd.DataFrame)
    def dropna(self, how=how_default, subset=None, thresh=thresh_default):
        if not PANDAS_GT_150 and how is not no_default and thresh is not no_default:
            raise ValueError(
                "You cannot set both the how and thresh arguments at the same time."
            )

         if how is no_default:
            how = "any"  
          
        return self.map_partitions(
            M.dropna, how=how, subset=subset, thresh=thresh, enforce_metadata=False
        )
```

Reply:
Yeah, I'd also drop the `not PANDAS_GT_150` part and just always raise that error appropriately